File: Insertion Sort.c
#include <stdio.h>

	void printarray(int array[], int size)

	{

	int i;

	for(i=0;i<size;i++)

	{

	printf("%d ", array[i]);

	}

	printf("\n");

	}

	void insertion_sort(int array[], int size)

	{

	int step;

	for (step = 1; step < size; step++)

	{

	int key = array[step];

	int j = step - 1;

	while (key < array[j] && j >= 0)

	{

	array[j + 1] = array[j];

	--j;

	}

	array[j + 1] = key;

	}

	}

	int main()

	{

	int a[25], i, n;

	printf("\nName : ***Aditya Panditrao*** ");

	printf("\n Enter the number of elements in an array: ");

	scanf("%d", &n);

	printf("\n Enter the elements of array: ");

	for(i = 0; i < n; i++)

	scanf("%d", &a[i]);

	insertion_sort(a, n);

	printf("Sorted array in ascending order:\n");

	printarray(a, n);

	}

File: InsertionSort.py
def insertionsort(array):
    print(array)
    
    for i in range(0,len(array)):
        b=i 
        while b > 0 and array[b]<array[b-1]:
            array[b-1],array[b]=array[b],array[b-1]
            b -=1 
            
    print(array)    
        
            
array =[1,6,7,8,3,4]
insertionsort(array)
